files/nagios/plugins/check_openstack_interface.py

```diff
@@ -0,0 +1,245 @@
+#!/usr/bin/env python3
+
+# Copyright (C) 2021 Canonical Ltd.
+
+# Authors:
+#   Robert Gildein <robert.gildein@canonical.com>
+
+import argparse
+import os
+from configparser import ConfigParser
+
+import openstack
+from nagios_plugin3 import try_check, UnknownError, CriticalError, WarnError
+
+
+SEPARATOR = ","
+OK_MESSAGE = "{}/{} passed"
+WARNING_MESSAGE = "{}/{} in UNKNOWN"
+DOWN_MESSAGE = "{}/{} are DOWN"
+NOT_FOUND_MESSAGE = "{}/{} were not found"
+NAGIOS_STATUS_OK = 0
+NAGIOS_STATUS_WARNING = 1
+NAGIOS_STATUS_CRITICAL = 2
+NAGIOS_STATUS_UNKNOWN = 3
+NAGIOS_STATUS = {
+    NAGIOS_STATUS_OK: "OK",
+    NAGIOS_STATUS_WARNING: "WARNING",
+    NAGIOS_STATUS_CRITICAL: "CRITICAL",
+    NAGIOS_STATUS_UNKNOWN: "UNKNOWN",
+}
+INTERFACE = {
```

Comment:
`INTERFACE` is a bit of a confusing name in this context because i first thought it was "network interface/ port"

perhaps you could use the name `RESOURCE`

but I realize that could be impactful to this PR. 

So it's just a question/thought exercise really

Reply:
The name comes from a time when only the network and port were checked, and I neglected renaming.
You are right and I'll rename it, also the name of file.